### PR TITLE
fix(menu): should not make a selection on right click

### PR DESCRIPTION
**Pull request by TarunAdobe**
## Description

Menu should not make a selection on right click. This was causing issues in case the consumers were using `contextmenu` to show the menu.

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_

    1. Go here
    2. Do this

-   [ ] Did it pass in Desktop?
-   [ ] Did it pass in Mobile?
-   [ ] Did it pass in iPad?

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by TarunAdobe** [comment]
Talking to the design and accessibility change to confirm and get a better clarity at this behaviour.

**Comment by Rajdeepc** [comment]
> Talking to the design and accessibility change to confirm and get a better clarity at this behaviour.

This is a miss! I checked the WCAG guidelines on the menu behaviour, we should definitely not allow right click event on a `sp-menu`.

**Comment by TarunAdobe** [comment]
> > Talking to the design and accessibility change to confirm and get a better clarity at this behaviour.
> 
> This is a miss! I checked the WCAG guidelines on the menu behaviour, we should definitely not allow right click event on a `sp-menu`.

Nice. And how do we feel about mouse middle click? Should the `sp-menu` component only make a selection on mouse left click and ignore everything else or not?

**Comment by Rajdeepc** [comment]
> > > Talking to the design and accessibility change to confirm and get a better clarity at this behaviour.
> > 
> > 
> > This is a miss! I checked the WCAG guidelines on the menu behaviour, we should definitely not allow right click event on a `sp-menu`.
> 
> Nice. And how do we feel about mouse middle click? Should the `sp-menu` component only make a selection on mouse left click and ignore everything else or not?

Yes it should only make a selection with a left mouse click, middle and right mouse clicks should not attach a event listener to the target. I checked the a11y guidelines, only left click is part of the guidelines!

**Comment by TarunAdobe** [comment]
> > > > Talking to the design and accessibility change to confirm and get a better clarity at this behaviour.
> > > 
> > > 
> > > This is a miss! I checked the WCAG guidelines on the menu behaviour, we should definitely not allow right click event on a `sp-menu`.
> > 
> > 
> > Nice. And how do we feel about mouse middle click? Should the `sp-menu` component only make a selection on mouse left click and ignore everything else or not?
> 
> Yes it should only make a selection with a left mouse click, middle and right mouse clicks should not attach a event listener to the target. I checked the a11y guidelines, only left click is part of the guidelines!

Cool. Updated the code.

**Comment by Rajdeepc** [comment]
Current implementation looks good, but It would be great if we can add some tests so that it doesn't leak for any future changes!

File: packages/menu/test/menu.test.ts [hunk 1]
```diff
@@ -500,6 +502,52 @@ describe('Menu', () => {
...
+                    options: {
+                        button: 'right',
+                    },
+                },
+            ],
+        });
+        await elementUpdated(el);
+        await elementUpdated(secondItem);
+        await aTimeout(150);
+        expect(changeSpy.callCount, 'no change').to.equal(0);
```
[the remark this answers is not in the document]
Comment (TarunAdobe) [reply packages/menu/test/menu.test.ts:569]: Yess!! done.

File: packages/menu/README.md [hunk 2]
```diff
@@ -160,3 +160,5 @@ When `selects` is set to `inherit`, the `<sp-menu>` element will allow its `<sp-
 ## "change" event
 
 Whether `<sp-menu>` carries a selection or not, when one of the `<sp-menu-item>` children that it manages is activated the `<sp-menu>` element will dispatch a `change` event. At dispatch time, even when a selection is not held due to the absence of the `selects` attribute, the `value` of the `<sp-menu>` will correspond to the `<sp-menu-item>` that was activated. When the `selects` attribute is present, this `value` will be persisted beyond the lifecycle of the `change` event. When `selects="multiple"` the values of multiple items will be comma separated unless otherwise set via the `value-separator` attribute.
+
+Note: The `change` event is only dispatched on a left mouse click or a keyboard activation. Right/Middle mouse clicks will not dispatch the `change` event.
```
Comment (Rajdeepc) [inline packages/menu/README.md:164]: Keyboard activation is very generic, can we please mention the key?

Comment (TarunAdobe) [reply packages/menu/README.md]: Done!

**Review by rubencarvalho — APPROVED** [review]

**Review by Rajdeepc — APPROVED** [review]

**Pull request merged**